File: frontend/app_flowy/packages/appflowy_board/lib/src/utils/log.dart
import 'package:flutter/material.dart';

// ignore: constant_identifier_names
const DART_LOG = "Dart_LOG";

class Log {
  static const enableLog = false;

  static void info(String? message) {
    if (enableLog) {
      debugPrint('AppFlowyBoard: ℹ️[Info]=> $message');
    }
  }

  static void debug(String? message) {
    if (enableLog) {
      debugPrint(
          'AppFlowyBoard: 🐛[Debug] - ${DateTime.now().second}=> $message');
    }
  }

  static void warn(String? message) {
    if (enableLog) {
      debugPrint(
          'AppFlowyBoard: 🐛[Warn] - ${DateTime.now().second} => $message');
    }
  }

  static void trace(String? message) {
    if (enableLog) {
      debugPrint(
          'AppFlowyBoard: ❗️[Trace] - ${DateTime.now().second}=> $message');
    }
  }

  static void error(String? message) {
    debugPrint('AppFlowyBoard: ❌[Error] - ${DateTime.now().second}=> $message');
  }
}

File: frontend/appflowy_tauri/src/appflowy_app/utils/log.ts
export class Log {
  static error(...msg: unknown[]) {
    console.error(...msg);
  }
  static info(...msg: unknown[]) {
    console.info(...msg);
  }

  static debug(...msg: unknown[]) {
    console.debug(...msg);
  }

  static trace(...msg: unknown[]) {
    console.trace(...msg);
  }

  static warn(...msg: unknown[]) {
    console.warn(...msg);
  }
}
